source/frontend/src/utils/fileUtils.ts

```diff
@@ -0,0 +1,42 @@
+import moment from 'moment';
+
+import { ApiGen_Base_BaseAudit } from '@/models/api/generated/ApiGen_Base_BaseAudit';
+import { ApiGen_Concepts_FileChecklistItem } from '@/models/api/generated/ApiGen_Concepts_FileChecklistItem';
+
+export function sortByDisplayOrder(
+  a: ApiGen_Concepts_FileChecklistItem,
+  b: ApiGen_Concepts_FileChecklistItem,
+) {
+  return (a.itemType?.displayOrder ?? 0) - (b.itemType?.displayOrder ?? 0);
+}
+
+export function lastModifiedBy(
+  array: ApiGen_Base_BaseAudit[] = [],
+): ApiGen_Base_BaseAudit | undefined {
+  let lastModified: ApiGen_Base_BaseAudit | undefined = undefined;
+
+  for (const item of array) {
+    if (moment(item.appLastUpdateTimestamp).isBefore(moment('1900-01-01T00:00:00.000Z'))) {
+      continue;
+    }
+    if (lastModified === undefined) {
+      lastModified = item;
+      continue;
+    }
+    if (moment(item.appLastUpdateTimestamp).isAfter(moment(lastModified.appLastUpdateTimestamp))) {
+      lastModified = item;
+    }
+  }
+
+  return lastModified;
+}
+
+export function isDefaultState(checklistItem: ApiGen_Concepts_FileChecklistItem): boolean {
+  return (
+    checklistItem.statusTypeCode?.id === 'INCOMP' &&
```

Comment:
at some point we'll probably want a story to replace all of these with their generated enum values. Don't get me wrong though, this PR is large enough as it is.

Reply:
agreed! note that this change just moves that existing logic to this new file